single_kernel_mongo/managers/backups.py

```diff
@@ -141,6 +153,89 @@ def backup_id(self) -> str:
     def backup_id(self, value: str):
         self._backup_id = value
 
+    def _get_bucket_resource(self, credentials: dict[str, str]) -> Bucket:
+        """Get the Bucket resource from the s3 connection.
+
+        Returns:
+            Bucket: the s3 bucket for uploading/downloading backups
+        """
+        s3_resource = boto3.resource(
+            "s3",
+            region_name=credentials.get("region"),
+            endpoint_url=credentials["endpoint"],
+            aws_access_key_id=credentials["access-key"],
+            aws_secret_access_key=credentials["secret-key"],
+            config=BotoConfig(
+                # https://github.com/boto/boto3/issues/4400#issuecomment-2600742103
+                request_checksum_calculation="when_required",
+                response_checksum_validation="when_required",
+            ),
+            verify=(TRUST_STORE_PATH / TrustStoreFiles.PBM.value)
+            if credentials.get("tls-ca-chain")
+            else True,
+        )
+
+        return s3_resource.Bucket(credentials["bucket"])
+
+    @retry(
+        stop=stop_after_attempt(5),
+        retry=retry_if_exception_type(ConnectTimeoutError),
+        wait=wait_fixed(5),
+        reraise=True,
+    )
+    def create_bucket(self, credentials: dict[str, str]) -> None:
+        """Create bucket if it does not exist yet."""
+        region = credentials.get("region")
+        bucket_name = credentials["bucket"]
+
+        if tls_ca_chain := credentials.get("tls-ca-chain", None):
+            with open(TRUST_STORE_PATH / TrustStoreFiles.PBM.value, mode="w") as fd:
+                # boto3 client will need the certificate on the node that runs the command
+                fd.write("\n".join(tls_ca_chain))
+
+        bucket = self._get_bucket_resource(credentials)
+
+        try:
+            bucket.meta.client.head_bucket(Bucket=bucket_name)
+            logger.info(f"Using existing bucket {bucket_name}")
+            exists = True
```

Comment:
this is not needed, we can return here, and omit the need for this variable altogether